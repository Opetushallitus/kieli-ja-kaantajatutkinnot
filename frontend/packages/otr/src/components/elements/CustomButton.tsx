import { Button, ButtonProps } from '@mui/material';
import { FC } from 'react';
import { LinkProps } from 'react-router-dom';

type CustomButtonProps = ButtonProps &
  Partial<Pick<LinkProps, 'target' | 'rel'>>;

export const CustomButton: FC<CustomButtonProps> = ({ ...props }) => {
  return <Button {...props} aria-disabled={props.disabled} />;
};
