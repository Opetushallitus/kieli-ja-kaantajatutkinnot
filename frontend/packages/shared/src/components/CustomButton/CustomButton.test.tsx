import renderer from 'react-test-renderer';

import { CustomButton } from './CustomButton';

describe('CustomButton', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<CustomButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
