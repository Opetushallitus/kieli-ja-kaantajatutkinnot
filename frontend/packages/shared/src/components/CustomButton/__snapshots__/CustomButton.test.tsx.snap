// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton should render correctly 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
/>
`;
